src/proto/mgmt/pool.proto

```diff
@@ -37,23 +37,29 @@ message FaultDomain {
 
 // PoolCreateReq supplies new pool parameters.
 message PoolCreateReq {
-	uint64 scmbytes = 1; // SCM size in bytes
-	uint64 nvmebytes = 2; // NVMe size in bytes
-	repeated uint32 ranks = 3; // target ranks
-	uint32 numsvcreps = 4; // desired number of pool service replicas
-	string user = 5; // formatted user e.g. "bob@"
-	string usergroup = 6; // formatted group e.g. "builders@"
-	string uuid = 7; // UUID for new pool, generated on the client
-	string sys = 8; // DAOS system identifier
-	repeated string acl = 9; // Access Control Entries in short string format
-	repeated FaultDomain faultDomains = 10; // Fault domains in system
+	string uuid = 1; // UUID for new pool, generated on the client
+	string name = 2; // Unique name for pool (optional)
+	string sys = 3; // DAOS system identifier
+	string user = 4; // formatted user e.g. "bob@"
+	string usergroup = 5; // formatted group e.g. "builders@"
+	repeated string acl = 6; // Access Control Entries in short string format
+	repeated FaultDomain faultDomains = 7; // Fault domains in system
+	uint32 numsvcreps = 8; // desired number of pool service replicas
+	uint64 totalbytes = 9; // Total pool size in bytes (auto config)
```

Comment:
would be cleaner to have oneof config config {auto, manual} with the relevant sub-messages defined

Reply:
The thing is, the scm/nvme fields are always filled in when sending the request down via dRPC, so it's not really an either/or situation.